core/core.go

```diff
@@ -72,13 +73,23 @@ func Run() error {
 
 	// store.Singleton is a thread-safe abstraction to load and store and string,
 	// provided by mesos-go.
-	// We also make sure that a log message is printed when the FrameworkID changes.
+	// We also make sure that a log message is printed with the FrameworkID.
 	fidStore := store.DecorateSingleton(
 		store.NewInMemorySingleton(),
 		store.DoSet().AndThen(func(_ store.Setter, v string, _ error) error {
-			log.WithField("frameworkId", v).Debug("generated new frameworkId")
+			// Store Mesos Framework ID to configuration.
+			err = the.ConfSvc().NewMesosFID(v)
+			if err != nil {
+				log.WithField("error", err).Error("cannot write to configuration")
+			}
+			log.WithField("frameworkId", v).Debug("frameworkId")
```

Comment:
Am I right in assuming that the `AndThen` here runs after every `Set` of the `fidStore`, the first occasion of which happens immediately after subscription when an empty `fidStore` receives a FID from Mesos?

Reply:
That's correct.